pkg/splunk/controller/statefulset.go

```diff
@@ -326,7 +326,7 @@ func DeleteReferencesToAutomatedMCIfExists(ctx context.Context, client splcommon
 
 //isCurrentCROwner returns true if current CR is the ONLY owner of the automated MC
 func isCurrentCROwner(cr splcommon.MetaObject, currentOwners []metav1.OwnerReference) bool {
-	return reflect.DeepEqual(currentOwners[0], splcommon.AsOwner(cr, false))
+	return reflect.DeepEqual(currentOwners[0].UID, cr.GetUID())
```

Comment:
Same comment as above. Maybe we can create function to compare the four fields - kind, name, UID and controller.

Reply:
we do not need controller, we just need  UID. UID will uniquely identify the resource. please [refer](https://kubernetes.io/docs/concepts/overview/working-with-objects/names/#:~:text=A%20Kubernetes%20systems%2Dgenerated%20string,(also%20known%20as%20UUIDs).).